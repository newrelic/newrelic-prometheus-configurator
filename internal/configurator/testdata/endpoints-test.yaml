kubernetes:
  jobs:
    # Job with kubernetes sd config not populated and with extra_label_config
    - job_name_prefix: default
      target_discovery:
        endpoints: true

      # Added extra label to test that we place properly already existing configs.
      extra_relabel_config:
      - action: drop
        regex: (.+)
        source_labels:
        - test_label

    # Job with kubernetes sd config actually populated
    - job_name_prefix: default-2
      target_discovery:
        endpoints: true
        additional_config:
          kubeconfig_file: test_file
          namespaces:
            own_namespace: false
            names:
              - namespace1
                namespace2
          selectors:
            - role: endpoints
              label: test_label
              field: app=test
          attach_metadata:
            node: false

newrelic_remote_write:
  license_key: nrLicenseKey
