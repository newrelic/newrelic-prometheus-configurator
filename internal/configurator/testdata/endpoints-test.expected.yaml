scrape_configs:
  - job_name: kubernetes-job-endpoints
    kubernetes_sd_configs:
      - role: endpoints
    relabel_configs:
      - source_labels: [ __meta_kubernetes_pod_phase ]
        action: drop
        regex: Pending|Succeeded|Failed|Completed
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scheme]
        action: replace
        target_label: __scheme__
        regex: (https?)
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [ __address__, __meta_kubernetes_service_annotation_prometheus_io_port ]
        action: replace
        regex: (.+?)(?::\d+)?;(\d+)
        target_label: __address__
        replacement: $1:$2
      - action: labelmap
        regex: __meta_kubernetes_service_annotation_prometheus_io_param_(.+)
        replacement: __param_$1
      - action: labelmap
        regex: __meta_kubernetes_service_label_(.+)
      - source_labels: [__meta_kubernetes_namespace]
        action: replace
        target_label: namespace
      - source_labels: [__meta_kubernetes_service_name]
        action: replace
        target_label: service
      - source_labels: [ __meta_kubernetes_pod_node_name ]
        action: replace
        target_label: node

        # Extra relabel config
      - source_labels: [ test_label ]
        action: drop
        regex: (.+)

  - job_name: kubernetes-job-2-endpoints
    kubernetes_sd_configs:
      - role: endpoints
        kubeconfig_file: test_file
        namespaces:
          own_namespace: false
          names:
          - namespace1
            namespace2
        selectors:
        - role: endpoints
          label: test_label
          field: app=test
        attach_metadata:
          node: false
    relabel_configs:
      - source_labels: [ __meta_kubernetes_pod_phase ]
        action: drop
        regex: Pending|Succeeded|Failed|Completed
      - source_labels: [ __meta_kubernetes_service_annotation_prometheus_io_scheme ]
        action: replace
        target_label: __scheme__
        regex: (https?)
      - source_labels: [ __meta_kubernetes_service_annotation_prometheus_io_path ]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [ __address__, __meta_kubernetes_service_annotation_prometheus_io_port ]
        action: replace
        regex: (.+?)(?::\d+)?;(\d+)
        target_label: __address__
        replacement: $1:$2
      - action: labelmap
        regex: __meta_kubernetes_service_annotation_prometheus_io_param_(.+)
        replacement: __param_$1
      - action: labelmap
        regex: __meta_kubernetes_service_label_(.+)
      - source_labels: [ __meta_kubernetes_namespace ]
        action: replace
        target_label: namespace
      - source_labels: [ __meta_kubernetes_service_name ]
        action: replace
        target_label: service
      - source_labels: [ __meta_kubernetes_pod_node_name ]
        action: replace
        target_label: node

remote_write:
  - url: https://metric-api.newrelic.com/prometheus/v1/write
    authorization:
      credentials: nrLicenseKey
